website/src/pages/dictionary/Schema/index.js

```diff
@@ -0,0 +1,104 @@
+import React, { useState } from 'react';
+import Table from '@icgc-argo/uikit/Table';
+import TagButton, { TAG_TYPES } from './TagButton';
+import styles from './styles.module.css';
+import Tag from '@icgc-argo/uikit/Tag';
+import CodeList from './CodeList';
+import Regex from './Regex';
+const formatFieldType = value => {
+  switch (value) {
+    case 'string':
+      return 'TEXT';
+    default:
+      return value.toUpperCase();
+  }
+};
+
+const FieldDescription = ({ name, description }) => (
+  <div className={styles.fieldDescription}>
+    <div className={styles.name}>{name}</div>
+    <div>{description}</div>
+  </div>
+);
+
+const FieldsTag = ({ fieldCount }) => (
+  <Tag className={styles.fieldsTag}>{`${fieldCount} Field${fieldCount > 1 ? 's' : ''}`}</Tag>
+);
+
+const Schema = ({ schema, key }) => {
+  const fields = schema.fields.map(field => {
+    /**
+     * need to pass in state for Cell rendering
+     * react-table rerenders everything, change shape of codelist to pass in state
+     */
+    const { restrictions: { codeList = null } = {} } = field;
+
+    if (codeList) {
+      const [isExpanded, setExpanded] = useState(false);
+      const newCodeList = {};
+      newCodeList.values = Array.isArray(codeList) ? codeList : codeList.values;
+      newCodeList.isExpanded = isExpanded;
+      newCodeList.setExpanded = setExpanded;
+      return { ...field, restrictions: { codeList: newCodeList } };
+    }
+    return field;
+  });
+
+  const cols = [
+    {
+      Header: 'Field & Description',
+      id: 'fieldDescription',
+      accessor: ({ name, description }) => (
+        <FieldDescription name={name} description={description} />
+      ),
+    },
+    { Header: 'Data Tier' },
+    {
+      Header: 'Attributes',
+      id: 'attributes',
+      accessor: ({ restrictions }) =>
+        restrictions && restrictions.required && <TagButton type={TAG_TYPES.required} />,
+    },
+    { Header: 'Type', id: 'valueType', accessor: ({ valueType }) => formatFieldType(valueType) },
+    {
+      Header: 'Permissible Values',
+      id: 'permissibleValues',
+      accessor: 'restrictions',
+      Cell: ({ original: { restrictions = {} } }) => {
+        const { regex = null, codeList = null } = restrictions;
+        if (regex) {
+          return <Regex regex={regex} />;
+        } else if (codeList) {
+          return <CodeList codeList={codeList} />;
```

Comment:
maybe it's better to have `CodeList` receive a curried `onToggle` rather than including `setExpanded` as part of `fields`. 

Naively, it could look something like: `<CodeList onToggle={onCodelistExpandToggle(field)} />` where `onCodelistExpandToggle` calls `setExpanded`. 

That way the declaration and invocation of `setExpanded` are both in the same component rather than being scattered between `Schema` and `CodeList`. Also keeping the `fields` serializable (only containing primitives) seems easier to implement future requirements that require serialization (tsv export maybe?)

Reply:
fixed https://github.com/icgc-argo/argo-docs/pull/35/commits/c81b681dca22c52e4aca60bd7f60df6e32e3eb31